librz/analysis/arch/8051/8051_il.c

```diff
@@ -0,0 +1,461 @@
+		set_add_carry(VAL_A, VARL("arg1"), VARL("res")),
+		set_a(VARL("res")));
+}
+
+static RzILOpEffect *i_subb(I8051Op *op) {
+	rz_return_val_if_fail(op && op->argv[0], NULL);
+	return SEQ4(SETL("arg1", get_any(op->argv[1])),
+		SETL("res", SUB3(VAL_A, VARL("arg1"), VAL_CY8)),
+		set_sub_curry(VAL_A, VARL("arg1"), VARL("res")),
+		set_a(VARL("res")));
+}
+static RzILOpEffect *i_div(I8051Op *op) {
+	return SEQ6(SETL("@a", get_any(op->argv[0])), SETL("@b", get_any(op->argv[1])),
+		set_a(DIV(VARL("@a"), VARL("@b"))), set_b(MOD(VARL("@a"), VARL("@b"))),
+		set_reg(I8051_CY, IL_FALSE), set_reg(I8051_OV, IS_ZERO(VARL("@b"))));
+}
+static RzILOpEffect *i_mul(I8051Op *op) {
+	return SEQ6(SETL("v", DIV(CAST(16, IL_FALSE, VAL_A), CAST(16, IL_FALSE, VAL_B))),
+		SETL("mbv", HI_BYTE(VARL("v"))),
+		set_a(LO_BYTE(VARL("v"))), set_b(VARL("mbv")),
+		set_reg(I8051_CY, IL_FALSE), set_reg(I8051_OV, NON_ZERO(VARL("mbv"))));
+}
+
+static RzILOpEffect *i_dec(I8051Op *op) {
+	rz_return_val_if_fail(op && op->argv[0], NULL);
+	return SEQ2(SETL("arg0", get_any(op->argv[0])), set_any(op->argv[0], SUB(VARL("arg0"), U8(1))));
+}
+static RzILOpEffect *i_inc(I8051Op *op) {
+	rz_return_val_if_fail(op && op->argv[0], NULL);
+	RzILOpPure *one = op->opcode == 0xa3 ? U16(1) : U8(1);
+	return SEQ2(SETL("arg0", get_any(op->argv[0])), set_any(op->argv[0], ADD(VARL("arg0"), one)));
+}
+
+static RzILOpEffect *i_clr(I8051Op *op) {
+	rz_return_val_if_fail(op, NULL);
+	return setb_any(op->argv[0], false);
+}
+static RzILOpEffect *i_setb(I8051Op *op) {
+	rz_return_val_if_fail(op, NULL);
+	return setb_any(op->argv[0], true);
+}
+static RzILOpEffect *i_cpl(I8051Op *op) {
+	rz_return_val_if_fail(op, NULL);
+	RzILOpPure *v = (op->argv[0]->mode == I8051_ADDRESSING_REGISTER && i8051_register_is_bit[op->argv[0]->d.reg]) || op->argv[0]->mode == I8051_ADDRESSING_BIT
+		? INV(get_any(op->argv[0]))
+		: LOGNOT(get_any(op->argv[0]));
+	return set_any(op->argv[0], v);
+}
+static RzILOpEffect *i_anl(I8051Op *op) {
+	rz_return_val_if_fail(op && op->argv[0] && op->argv[1], NULL);
+	if (op->argv[0]->mode == I8051_ADDRESSING_BIT || op->argv[1]->mode == I8051_ADDRESSING_BIT) {
+		return set_any(op->argv[0], AND(get_any(op->argv[0]), get_any(op->argv[1])));
+	}
+	return set_any(op->argv[0], LOGAND(get_any(op->argv[0]), get_any(op->argv[1])));
+}
+static RzILOpEffect *i_orl(I8051Op *op) {
+	rz_return_val_if_fail(op && op->argv[0] && op->argv[1], NULL);
+	if (op->argv[0]->mode == I8051_ADDRESSING_BIT || op->argv[1]->mode == I8051_ADDRESSING_BIT) {
+		return set_any(op->argv[0], OR(get_any(op->argv[0]), get_any(op->argv[1])));
+	}
+	return set_any(op->argv[0], LOGOR(get_any(op->argv[0]), get_any(op->argv[1])));
+}
+static RzILOpEffect *i_xrl(I8051Op *op) {
+	rz_return_val_if_fail(op && op->argv[0] && op->argv[1], NULL);
+	if (op->argv[0]->mode == I8051_ADDRESSING_BIT || op->argv[1]->mode == I8051_ADDRESSING_BIT) {
+		return set_any(op->argv[0], XOR(get_any(op->argv[0]), get_any(op->argv[1])));
```

Comment:
Please add an asm test to cover this case.

Reply:
I checked and found that this case does not happen